### PR TITLE
hwloc: added missing ncurses dependency

### PR DESCRIPTION
**Pull request by westernmagic**

### CODE REVIEW

**Comment by adamjstewart** [comment]
@bgoglin can you review this PR?

**Comment by westernmagic** [comment]
Would appreciate a review @bgoglin

**Comment by bgoglin** [comment]
I am not sure it should be a strong dependency. lstopo/ascii should build fine without it, but the textual output is black and white, etc.

What's important is actually libtermcap and term.h. Once hwloc find those, it enables the lstopo/ascii backend to get colors in the textual outputgraphics. And that code assumes either curses.h or ncurses.h is available. It seems to me that termcap is included in ncurses. If that's true for the spack package, then this PR is OK.

File: var/spack/repos/builtin/packages/hwloc/package.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ class Hwloc(AutotoolsPackage):
     depends_on('libxml2', when='+libxml2')
     depends_on('cairo', when='+cairo')
     depends_on('numactl', when='@:1.11.11 platform=linux')
+    depends_on('ncurses')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/hwloc/package.py:88]:
```suggestion
    depends_on('ncurses')
    depends_on('ncurses ~termlib')
```
Currently hwloc does not detect `libtinfo.so`, so you have to disable `termlib`

**Comment by haampie** [comment]
hwloc's detection for ncurses is a bit flawed: it tries to compile `tparm(NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0)` and tries to links against either `ncurses`, `curses`, `termcap`, `ncursesw` or `cursesw`. With spack's default ncurses it is only defined in `tinfo`. So should be patched upstream (https://github.com/open-mpi/hwloc/pull/417).

For now I guess `depends_on('ncurses ~termlib')` will do.

@bgoglin, it's an optional dependency, but it is automatically detected, so it could end up linking to a system lib, which is not great.

**Comment by msimberg** [comment]
Any update on this? I'm hitting this problem as well when building hwloc.

**Comment by bgoglin** [comment]
FYI, the hwloc patch was included in hwloc 2.3. I don't know why this PR wasn't merged yet.

**Pull request closed without merging**